### PR TITLE
Really deep statement matching!

### PR DESCRIPTION
**Pull request by aryx**
This flatten the statements also of functions and classes
This fix issue #413

Test plan:
test file included

### CODE REVIEW

File: sgrep/tests/js/deep_deep_stmt1.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    function OldSchool () {
+        
+        this.test = function() {
+            console.log(x);
+        }
+        
+    }
+}
+
```
Comment (nbrahms) [inline sgrep/tests/js/deep_deep_stmt1.js:61]: I'd recommend a negative test cases along the lines of:

```javascript
const foo = () => {
  var x = 1;
}
console.log("hi")
```

with a pattern like
```
var $X = 1;
...
console.log("...");
```

**Review by inkz — COMMENTED** [review]
built branch locally and this example doesnt seem to work:
```
 var $X = 1;
...
console.log($X);
```

**UPD: my bad, was checking the wrong version!:innocent:**
actually everything works, I checked different complex examples from https://github.com/returntocorp/sgrep-rules/tree/develop/javascript/jwt-hardcode
and they all were found :+1:

**Review by inkz — DISMISSED** [review]
:+1:

**Comment by aryx** [comment]
@ievans found some huge perf regression in 0.4.9 with sgrep-rules, so given this is not even merged and this will make things even worse, we may want to wait to have a good perf regression infrastructure before merging this? (or setting really_deep_stmt flag to false by default).

**Pull request merged**